ts/components/services/NewServiceListItem.tsx

```diff
@@ -48,14 +94,25 @@ export default class NewServiceListItem extends React.PureComponent<Props> {
         ? I18n.t("global.remoteStates.notAvailable")
         : potService.value.service_name;
 
+    // whether last attempt to save the preferences failed
+    const profileVersion = pot
+      .toOption(this.props.profile)
+      .mapNullable(_ => (_.has_profile ? _.version : null))
+      .getOrElse(0 as NonNegativeInteger);
+
     return (
       <ListItemComponent
         title={serviceName}
         subTitle={inboxEnabledLabel}
         hasBadge={!this.props.isRead}
         onPress={onPress}
+        onLongPress={this.props.onLongPress}
         hideSeparator={this.props.hideSeparator}
         style={styles.listItem}
+        onSwitchValueChanged={onItemSwitchValueChanged}
+        switchValue={pot.isSome(enabledChannels) && enabledChannels.value.inbox}
+        keySwitch={`switch-service-${profileVersion}`}
```

Comment:
The key is useful to identify an item in a list. This is already done in ServiceSectionListComponent (_keyExtractor_)
In this is case you don't need a key to identify the switch
Anyway if you want to pass an identifier you can build a valid one using the **_service id_** 

```typescript
private getServiceKey = (
    potService: pot.Pot<ServicePublic, Error>
  ): string => {
    return pot.getOrElse(
      pot.map(
        potService,
        service => `${service.service_id}-${service.version || 0}`
      ),
      `service-switch`
    );
  };
```

note `profileVersion` will be always the same value for all switches

Reply:
Ok, I fix it. Thanks